### PR TITLE
Add information on version tag differences.

### PR DESCRIPTION
**Pull request by VersusFacit**
## Description & motivation
Paired documentation upgrade for [this PR here](https://github.com/dbt-labs/dbt-core/pull/4816).

For more background, see the recent Slack discussions on this.

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [X] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

### CODE REVIEW

File: website/docs/reference/project-configs/version.md [hunk 1]
```diff
@@ -10,7 +15,22 @@ version: version
...
+```yml
+version: 2  # Only 2 is accepted by current and recent versions of dbt.
+```
+ 
+</File>
+
+### Definition
+A control tag that informs how dbt processes property files. For more on why 2 is required, see property file [FAQs](https://docs.getdbt.com/reference/configs-and-properties#faqs).
+
+For more on property files, see their general [documentation](https://docs.getdbt.com/reference/configs-and-properties#where-can-i-define-properties) on the same page.
```
Comment (runleonarun) [inline website/docs/reference/project-configs/version.md]:
```suggestion
For more on property files, see their general [documentation](/reference/configs-and-properties#where-can-i-define-properties) on the same page.
```

File: website/docs/reference/project-configs/version.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 datatype: version
 required: True
 ---
+
+dbt projects have two distinct types of the `version` tags. This field has a different meaning depending on its location.
+
+## 1. In dbt_project.yml
```
Comment (runleonarun) [inline website/docs/reference/project-configs/version.md]: What do you think about removing the number and adding versions to this header?
```suggestion
## `dbt_project.yml` versions
```

File: website/docs/reference/project-configs/version.md [hunk 3]
```diff
@@ -10,7 +15,22 @@ version: version
 
 </File>
 
-## Definition
+### Definition
 The version of a dbt project. Note that while **this is a required parameter**, it is not currently meaningfully used by dbt.
 
 The version must be in a [semantic version](https://semver.org/) format, e.g. `1.0.0`
+
+## 2. In yml property files
```
Comment (runleonarun) [inline website/docs/reference/project-configs/version.md]:
```suggestion
##  `.yml` property file versions
```

File: website/docs/reference/project-configs/version.md [hunk 4]
```diff
@@ -2,6 +2,11 @@
 datatype: version
 required: True
 ---
+
+dbt projects have two distinct types of the `version` tags. This field has a different meaning depending on its location.
+
+## 1. In dbt_project.yml
+
```
Comment (runleonarun) [inline website/docs/reference/project-configs/version.md:9]:
```suggestion

The version tag in a `dbt_project` file represents the version of your dbt project. Although **this is a required parameter**, it is not currently meaningfully used by dbt. The version must be in a [semantic version](https://semver.org/) format, e.g. `1.0.0`. For more on Core versions, see "[About dbt Core versions](/docs/core-versions)."
```

File: website/docs/reference/project-configs/version.md [hunk 5]
```diff
@@ -10,7 +15,22 @@ version: version
 
 </File>
 
-## Definition
+### Definition
 The version of a dbt project. Note that while **this is a required parameter**, it is not currently meaningfully used by dbt.
```
Comment (runleonarun) [inline website/docs/reference/project-configs/version.md]: I moved this up to the top
```suggestion
```

File: website/docs/reference/project-configs/version.md [hunk 6]
```diff
@@ -10,7 +15,22 @@ version: version
...
 
 The version must be in a [semantic version](https://semver.org/) format, e.g. `1.0.0`
```
Comment (runleonarun) [inline website/docs/reference/project-configs/version.md:20]: I also moved this line up to the top
```suggestion
```

File: website/docs/reference/project-configs/version.md [hunk 7]
```diff
@@ -10,7 +15,22 @@ version: version
 
 </File>
 
-## Definition
+### Definition
```
Comment (runleonarun) [inline website/docs/reference/project-configs/version.md]:
```suggestion
```

File: website/docs/reference/project-configs/version.md [hunk 8]
```diff
@@ -10,7 +15,22 @@ version: version
...
 The version of a dbt project. Note that while **this is a required parameter**, it is not currently meaningfully used by dbt.
 
 The version must be in a [semantic version](https://semver.org/) format, e.g. `1.0.0`
+
+## 2. In yml property files
+
```
Comment (runleonarun) [inline website/docs/reference/project-configs/version.md:23]:
```suggestion

A version tag in a `.yml` property file provides the control tag, which informs how dbt processes property files. For more on why we require this tag, see property file [FAQs](https://docs.getdbt.com/reference/configs-and-properties#faqs).

For more on property files, see their general [documentation](https://docs.getdbt.com/reference/configs-and-properties#where-can-i-define-properties) on the same page.
```

File: website/docs/reference/project-configs/version.md [hunk 9]
```diff
@@ -10,7 +15,22 @@ version: version
...
+<File name='<any valid filename>.yml'>
+
+```yml
+version: 2  # Only 2 is accepted by current and recent versions of dbt.
+```
+ 
+</File>
+
+### Definition
```
Comment (runleonarun) [inline website/docs/reference/project-configs/version.md]: Moved this up
```suggestion
```

File: website/docs/reference/project-configs/version.md [hunk 10]
```diff
@@ -10,7 +15,22 @@ version: version
...
+A control tag that informs how dbt processes property files. For more on why 2 is required, see property file [FAQs](https://docs.getdbt.com/reference/configs-and-properties#faqs).
```
Comment (runleonarun) [inline website/docs/reference/project-configs/version.md]: I moved this up before the example.
```suggestion
```

File: website/docs/reference/project-configs/version.md (hunk 1 above)
Comment (runleonarun) [inline website/docs/reference/project-configs/version.md]: I also moved this up before the example.
```suggestion
```

**Review by runleonarun — COMMENTED** [review]
@VersusFacit This is SUCH valuable info. Thank you for getting it in the docs and doing the research for it. 

I have some minor requests. Slight wording suggestions but mostly moving the content before the example. Let me know what you think!

File: website/docs/reference/project-configs/version.md (hunk 2 above)
Comment (VersusFacit) [reply website/docs/reference/project-configs/version.md]: Oh yeah much preferred

File: website/docs/reference/project-configs/version.md (hunk 4 above)
Comment (VersusFacit) [reply website/docs/reference/project-configs/version.md:9]: Is this true? Judging from Jaffle shop and the existing version explanation, I'm pretty sure this version tag representation the version of the "dbt project" (not the core version). I directly edited and committed your suggestion.

File: website/docs/reference/project-configs/version.md (hunk 1 above)
Comment (VersusFacit) [reply website/docs/reference/project-configs/version.md]: I'm confused by this -- forgive my ignorance. What's the right formatting?

Comment (runleonarun) [reply website/docs/reference/project-configs/version.md]: @VersusFacit fixed now. Thanks for catching that! 😳

File: website/docs/reference/project-configs/version.md [hunk 11]
```diff
@@ -10,7 +16,20 @@ version: version
...
-## Definition
-The version of a dbt project. Note that while **this is a required parameter**, it is not currently meaningfully used by dbt.
 
 The version must be in a [semantic version](https://semver.org/) format, e.g. `1.0.0`
+
+##  `.yml` property file versions
+
+A version tag in a `.yml` property file provides the control tag, which informs how dbt processes property files. For more on why we require this tag, see property file [FAQs](https://docs.getdbt.com/reference/configs-and-properties#faqs).
+
+For more on property files, see their general [documentation](https://docs.getdbt.com/reference/configs-and-properties#where-can-i-define-properties) on the same page.
```
Comment (runleonarun) [inline website/docs/reference/project-configs/version.md]:
```suggestion
For more on property files, see their general [documentation](reference/configs-and-properties#where-can-i-define-properties) on the same page.
```

File: website/docs/reference/project-configs/version.md [hunk 12]
```diff
@@ -10,7 +16,20 @@ version: version
...
 </File>
 
-## Definition
-The version of a dbt project. Note that while **this is a required parameter**, it is not currently meaningfully used by dbt.
 
 The version must be in a [semantic version](https://semver.org/) format, e.g. `1.0.0`
+
+##  `.yml` property file versions
+
+A version tag in a `.yml` property file provides the control tag, which informs how dbt processes property files. For more on why we require this tag, see property file [FAQs](https://docs.getdbt.com/reference/configs-and-properties#faqs).
```
Comment (runleonarun) [inline website/docs/reference/project-configs/version.md]:
```suggestion
A version tag in a `.yml` property file provides the control tag, which informs how dbt processes property files. For more on why we require this tag, see property file [FAQs](reference/configs-and-properties#faqs).
```

**Review by runleonarun — APPROVED** [review]
I think these should work?

**Comment by VersusFacit** [comment]
@runleonarun I committed the changes and checked the links in the Netlify. Should be ready to go!

**Pull request merged**